spec/classes/epel_spec.rb

```diff
@@ -8,233 +8,117 @@
 require 'classes/shared_testing_debuginfo'
 
 describe 'epel' do
-  let :facts do
-    default_facts.merge(
-      operatingsystemrelease:    '8.0.1905',
-      operatingsystemmajrelease: '8'
-    )
-  end
-
-  it { is_expected.to create_class('epel') }
-  it { is_expected.to contain_class('epel::params') }
-
-  context "operatingsystem => #{default_facts[:operatingsystem]}" do
-    context 'operatingsystemmajrelease => 8' do
-      let :facts do
-        default_facts.merge(
-          operatingsystemrelease:    '8.0.1905',
-          operatingsystemmajrelease: '8'
-        )
-      end
-
-      it_behaves_like :base_8
-      it_behaves_like :gpgkey_8
-      it_behaves_like :epel_source_8
-      it_behaves_like :epel_debuginfo_8
-      it_behaves_like :epel_testing_8
-      it_behaves_like :epel_testing_source_8
-      it_behaves_like :epel_testing_debuginfo_8
-
-      context 'epel_baseurl => https://example.com/epel/8/x86_64' do
-        let(:params) do
-          {
-            epel_baseurl: 'https://example.com/epel/8/x86_64'
-          }
-        end
-
-        it { is_expected.to contain_yumrepo('epel').with(baseurl: 'https://example.com/epel/8/x86_64') }
-        it { is_expected.to contain_yumrepo('epel').with(mirrorlist: 'absent') }
-      end
-
-      context 'epel_mirrorlist => absent' do
-        let(:params) do
-          {
-            epel_mirrorlist: 'absent'
-          }
-        end
-
-        it { is_expected.to contain_yumrepo('epel').with(mirrorlist: 'absent') }
-      end
-
-      context 'operatingsystemmajrelease undef' do
```

Comment:
I've got no idea what the point of this was...
`operatingsystemmajrelease` is a core fact and should have never been unavailable.

Reply:
The module originally needed to support very old facter.
See https://github.com/voxpupuli/puppet-epel/commit/8880020e21bb021b4af3e4ba347c22f5b2dbfcdd and https://github.com/voxpupuli/puppet-epel/issues/27